smt/expr.h

```diff
@@ -61,6 +59,7 @@ class expr {
     std::swap(ptr, other.ptr);
   }
 
+  expr(Z3_ast ast);
```

Comment:
why do you need to expose this? I did a very shallow review of your Souper patch but it wasn't immediately obvious why would you need this,